src/pages/[platform]/build-a-backend/storage/storage-v5-to-v6-migration-guide/index.mdx

```diff
@@ -0,0 +1,1265 @@
+        expiresAt: Date;
+      }
+
+      // downloadData
+      DownloadDataOutput {
+        cancel: (message?: string) => void;
+        readonly state: TransferTaskState;
+        result: Promise<StorageDownloadDataOutput<Item>>;
+      }
+
+      StorageDownloadDataOutput<Item> {
+        key: string;
+        lastModified?: Date;
+        size?: number;
+        eTag?: string;
+        metadata?: Record<string, string>;
+        versionId?: string;
+        contentType?: string;
+        body: {
+          blob: () => Promise<Blob>;
+          json: () => Promise<any>;
+          text: () => Promise<string>;
+        };
+      }
+      ```
+
+    </div>
+  </Columns>
+</Accordion>
+
+### Get URL Only (Default)
+
+<BlockSwitcher>
+  <Block name="V6">
+    ```ts
+    import { getUrl } from 'aws-amplify/storage';
+
+    const handleGetUrl = async (key: string) => {
+      const url = await getUrl({
+        key,
+        options: {
+          validateObjectExistence: true
+        },
+      });
+    }
+    ```
+
+  </Block>
+  <Block name="V5">
+    ```ts
+    import { Storage } from 'aws-amplify';
+
+    const handleGetUrl = async (key: string) => {
+      const url = await Storage.get(key, {
+        validateObjectExistence: true
+      });
+    }
+    ```
+
+  </Block>
+</BlockSwitcher>
+
+### Download File Content to Memory
+
+<BlockSwitcher>
+  <Block name="V6">
+    ```ts
+    import { downloadData } from 'aws-amplify/storage';
+
+    const handleDownload = async (key: string) => {
+      const { body, eTag } = await downloadData({
+        key,
+        options: {
+          accessLevel: 'guest'
+        }
+      }).result;
+    }
+    ```
+
+  </Block>
+  <Block name="V5">
+    ```ts
+    import { Storage } from 'aws-amplify';
+
+    const handleDownload = async (key: string) => {
+      const { Body, ETag } = await Storage.get(key, {
+        download: true,
+        validateObjectExistence: true,
+        progressCallback: ({ loaded, total }) => {
+          // Progress implementation
+        }
+      });
+    }
+    ```
+
+  </Block>
+</BlockSwitcher>
+
+### Monitor Download Progress
+
+The `progressCallback` option has been replaced by `onProgress` in v6. The function input property names `loaded` and `total` now map to `transferredBytes` and `totalBytes`.
+
+<BlockSwitcher>
+  <Block name="V6">
+  ```
+  import { downloadData } from 'aws-amplify/storage';
+
+  const handleDownload = async (key: string) => {
+    const { body, eTag } = await downloadData({
+      key,
+      options: {
+        accessLevel: 'guest',
+        onProgress: ({ transferredBytes, totalBytes }) => {
+          // Progress implementation
+        }
+      }
+    }).result;
+  }
+  ```
+
+  </Block>
+  <Block name="V5">
+    ```
+    import { Storage } from 'aws-amplify';
+
+    const handleDownload = async (key: string) => {
+      const { Body, ETag } = await Storage.get(key, {
+        download: true,
+        validateObjectExistence: true
```

Comment:
Is this option extra?

Reply:
Yeah, basically unnecessary, wanted to demonstrate other parameters and how they map.